File: server-shared/src/main/java/eu/darken/porter/core/UserServiceConnection.java
package eu.darken.porter.core;

import android.os.IBinder;
import android.os.IInterface;
import android.os.RemoteException;

/**
 * Someone waiting to be told about a user service. Extends {@link IInterface} because
 * {@code RemoteCallbackList} keys its registrations on {@link IInterface#asBinder()}.
 */
public interface UserServiceConnection extends IInterface {

    void connected(IBinder service) throws RemoteException;

    void died() throws RemoteException;
}

File: server-shared/src/main/java/eu/darken/porter/core/UserServiceConnection.kt
package eu.darken.porter.core

import android.os.IBinder
import android.os.IInterface
import android.os.RemoteException

/**
 * Someone waiting to be told about a user service. Extends [IInterface] because
 * `RemoteCallbackList` keys its registrations on [IInterface.asBinder].
 */
interface UserServiceConnection : IInterface {

    @Throws(RemoteException::class)
    fun connected(service: IBinder)

    @Throws(RemoteException::class)
    fun died()
}
